plugins/woocommerce/tests/e2e-pw/global-setup.js

```diff
@@ -178,6 +181,62 @@ module.exports = async ( config ) => {
 		process.exit( 1 );
 	}
 
+
+	// While we're here, let's set HPOS according to the passed in ENABLE_HPOS env variable
+	// This was always being set to 'yes' after login in wp-env so this step ensures the
+	// correct value is set before we begin our tests
+	const hposSettingRetries = 5;
+	const api = new wcApi( {
+		url: baseURL,
+		consumerKey: process.env.CONSUMER_KEY,
+		consumerSecret: process.env.CONSUMER_SECRET,
+		version: 'wc/v3',
+	} );
+
+	for ( let i = 0; i < hposSettingRetries; i++ ) {
+		try {
+			if (ENABLE_HPOS === undefined || ENABLE_HPOS === '0') {
```

Comment:
But I noticed one side effect of the `if (ENABLE_HPOS === undefined || ENABLE_HPOS === '0')` logic: If you look at the [Runs E2E tests with COT enabled.](https://github.com/woocommerce/woocommerce/actions/runs/6377178749/job/17305370846?pr=40528#logs) for this PR, notice that HPOS was actually being turned off.

<img width="931" alt="Screenshot 2023-10-02 at 5 37 36 PM" src="https://github.com/woocommerce/woocommerce/assets/4509348/eb5defc2-c56c-4fd7-8aba-d88c14ff2e01">

That's because `ENABLE_HPOS` was set in the `Load docker images and start containers with COT enabled` step instead of in the `Run Playwright E2E tests.` step. So when it's time for the `Run Playwright E2E tests.` step to run, its ENABLE_HPOS is undefined:

<img width="870" alt="Screenshot 2023-10-02 at 5 33 20 PM" src="https://github.com/woocommerce/woocommerce/assets/4509348/e4e4dd1b-e890-4eb6-a6d7-e00d75356437">

I therefore think we should adjust the two `if` conditions such that when ENABLE_HPOS is undefined, then global-setup will do nothing, leaving the default HPOS setting enabled.

The two if conditions will then become:

```js
if (ENABLE_HPOS && ENABLE_HPOS === '0')

// ...

if (ENABLE_HPOS && ENABLE_HPOS === '1')
```
If we don't do this, once this PR is merged, all succeeding PR's would be tested against a disabled HPOS by the `pr-build-and-e2e-tests.yml` workflow since ENABLE_HPOS is undefined there too. Same with the daily and release tests -- their global setup will disable HPOS.

Afterwards, make the necessary adjustment in the `Runs E2E tests with COT enabled.` workflow to set ENABLE_HPOS to 0 in the `Run Playwright E2E tests.` step.

Do you agree with this approach?

BTW, I haven't given much thought about explicitly enabling HPOS when ENABLE_HPOS is undefined. I just feel like this approach could cause unintended consequences. I couldn't think of one right now. But I thought, if the tester didn't specify ENABLE_HPOS, then the tests should do nothing about it.